src/ComputeSharp.D2D1.SourceGenerators/ID2D1ShaderGenerator.CreateLoadBytecodeMethod.Syntax.cs

```diff
@@ -25,17 +27,17 @@ partial class LoadBytecode
         /// <param name="bytecodeInfo">The input bytecode info.</param>
         /// <param name="fixup">An opaque <see cref="Func{TResult}"/> instance to transform the final tree into text.</param>
         /// <returns>The resulting <see cref="MethodDeclarationSyntax"/> instance for the <c>BuildHlslSource</c> method.</returns>
-        public static MethodDeclarationSyntax GetSyntax(EmbeddedBytecodeInfo bytecodeInfo, out Func<SyntaxNode, string> fixup)
+        public static MethodDeclarationSyntax GetSyntax(EmbeddedBytecodeInfo bytecodeInfo, out Func<SyntaxNode, SourceText> fixup)
         {
             BlockSyntax block = GetShaderBytecodeBody(bytecodeInfo, out string? bytecodeLiterals);
 
             if (bytecodeLiterals is not null)
             {
-                fixup = tree => tree.ToFullString().Replace("__EMBEDDED_SHADER_BYTECODE", bytecodeLiterals);
+                fixup = tree => SourceText.From(tree.ToFullString().Replace("__EMBEDDED_SHADER_BYTECODE", bytecodeLiterals), Encoding.UTF8);
```

Comment:
Do you reckon there could be a way to skip the temporary string in this case? 🤔

Reply:
The implementation of `GetText` writes everything to a `StringBuilder` and return a `SourceText` which wraps that `StringBuilder`. But that `SourceText` implementation is internal. So we could explicitly create a `StringBuilder`, write the syntax into that, but there's no way to get a `SourceText` from that unless you create your own implementation, or use reflection and rely on the internal `Builder` property and replace the text there, which is ugly.

Reply:
It's unfortunate but I can't really think of anything that wouldn't require either reflection or a custom `SourceText` implementation.

Reply:
I mean we *could* write it to a `StringBuilder`, replace it in that `StringBuilder` and then do `ToString`, but I'm not sure whether that would be better at all. Time for a microbenchmark? 😄

Reply:
I'll be the contrarian and point out that I'm not sure that level of effort is worth it -- I'm using a lot of shaders in my codebase and performance in VS is quite good. ComputeSharp's generators don't seem to be causing any performance issues / lag / etc., AFAICT.

Reply:
I would keep this as is. I benchmarked the `StringBuilder` approach out of curiosity and it's not any faster.

Reply:
Well, great ahah
Plus we can always improve this specific bit as a follow up anyway

Reply:
@Sergio0694 You can of course make a request in roslyn to add a `Replace` method to `SourceText` if this bothers you 😄

Reply:
Actually, such a method would suck because it would have to copy the `StringBuilder` it wraps. The problem why the string approach is much faster here is that replacing is much faster on string than on `StringBuilder`.